util/cache/cache.go

```diff
@@ -41,11 +47,21 @@ func AddCacheFlagsToCmd(cmd *cobra.Command) func() (*Cache, error) {
 	sentinelAddresses := make([]string, 0)
 	sentinelMaster := ""
 	redisDB := 0
+	var defaultCacheExpiration time.Duration
+	var connectionStatusCacheExpiration time.Duration
+	var appStateCacheExpiration time.Duration
+	var repoCacheExpiration time.Duration
+	var oidcCacheExpiration time.Duration
 
 	cmd.Flags().StringVar(&redisAddress, "redis", "", "Redis server hostname and port (e.g. argocd-redis:6379). ")
 	cmd.Flags().IntVar(&redisDB, "redisdb", 0, "Redis database.")
 	cmd.Flags().StringArrayVar(&sentinelAddresses, "sentinel", []string{}, "Redis sentinel hostname and port (e.g. argocd-redis-ha-announce-0:6379). ")
 	cmd.Flags().StringVar(&sentinelMaster, "sentinelmaster", "master", "Redis sentinel master group name.")
+	cmd.Flags().DurationVar(&defaultCacheExpiration, "default-cache-expiration", 24*time.Hour, "Cache expiration default")
```

Comment:
This adds 5 new flags to both controller, api server and repo server. This is confusing because controller don't use `oidc-cache-expiration` flag, api server don't use `repo-cache-expiration` flag etc. I would use environment variables instead.

Reply:
@jessesuen thoughts?

Reply:
I'd note that the flags being exposed is because we have code in the repo server that it never uses and vice-versa. We could fix that.